File: telegram-bot-core/src/main/java/ru/tggc/telegrambotcore/registry/resolver/UserIdResolver.java
package ru.tggc.telegrambotcore.registry.resolver;

import org.springframework.stereotype.Component;
import ru.tggc.telegrambotcore.annotation.params.UserId;

import java.lang.reflect.Parameter;

@Component
public class UserIdResolver implements ParameterResolver<Long> {

    @Override
    public boolean supports(Parameter p) {
        return p.isAnnotationPresent(UserId.class);
    }

    @Override
    public Long resolve(Parameter parameter, HandlerCtx ctx) {
        return ctx.from().id();
    }
}

File: telegram-bot-core/src/main/kotlin/ru/tggc/telegrambotcore/registry/resolver/UserIdResolver.kt
package ru.tggc.telegrambotcore.registry.resolver

import org.springframework.stereotype.Component
import ru.tggc.telegrambotcore.annotation.params.UserId
import java.lang.reflect.Parameter

@Component
class UserIdResolver : ParameterResolver<Long?> {
    override fun supports(p: Parameter?): Boolean? = p?.isAnnotationPresent(UserId::class.java)

    override fun resolve(parameter: Parameter?, ctx: HandlerCtx): Long? = ctx.from?.id()
}
